llama_stack/distribution/routers/routing_tables.py

```diff
@@ -320,11 +318,38 @@ async def register_scoring_function(
 
 
 class EvalTasksRoutingTable(CommonRoutingTableImpl, EvalTasks):
-    async def list_eval_tasks(self) -> List[ScoringFnDefWithProvider]:
+    async def list_eval_tasks(self) -> List[EvalTask]:
         return await self.get_all_with_type("eval_task")
 
-    async def get_eval_task(self, name: str) -> Optional[EvalTaskDefWithProvider]:
+    async def get_eval_task(self, name: str) -> Optional[EvalTask]:
         return await self.get_object_by_identifier(name)
 
-    async def register_eval_task(self, eval_task_def: EvalTaskDefWithProvider) -> None:
-        await self.register_object(eval_task_def)
+    async def register_eval_task(
+        self,
+        eval_task_id: str,
+        dataset_id: str,
+        scoring_functions: List[str],
+        metadata: Optional[Dict[str, Any]] = None,
+        provider_id: Optional[str] = None,
+        provider_eval_task_id: Optional[str] = None,
```

Comment:
this is changed order - i like this better than asking for provider_resource_id before provider_id in the argument list

Reply:
done.